src/operators/CourseOperations.ts

```diff
@@ -0,0 +1,64 @@
+import Course, { CourseInterface } from '../models/course';
+import { v4 as uuidv4 } from 'uuid';
+import { checkUserId } from './index';
+
+const checkCourseUser = checkUserId(Course.get);
+
+export const create = async (course: CourseInterface) => {
+	try {
+		const hashKey = uuidv4();
+		course.id = hashKey;
+		await Course.create(course);
+		console.log(
+			'SUCCESS: Creation of course with id ' + course.id + ' successful.'
+		);
+	} catch (err) {
+		new Error('Failed to create course with id' + course.id);
```

Comment:
So this is not actually doing anything with the error. You need a `throw new Error(...)` or I think `throw Error(...)` also works.